src/js/services/workspace.js

```diff
@@ -42,23 +42,55 @@ export default App.extend({
   getDirectories() {
     return this.directories.clone();
   },
+  _getSharedWorkspaceIds(programs) {
```

Comment:
This is the crux of the logic of this PR.  However to really tested it aside from adding component tests we'll need the owner droplist UI in the following work

Reply:
considering the next step of this epic, there's a better way to do this